File: 876. Middle of the Linked List/count.cpp
class Solution{
public:
    ListNode* middleNode(ListNode* head) {
        ListNode *res = head;
        int count = 0;
        while (res != NULL) {
            count += 1;
            res = res->next;
        }
        res = head;
        for (int i = 0; i < count / 2; i++) {
            res = res->next;
        }
        return res;
    }
};

File: 876. Middle of the Linked List/count.py
class Solution:
    def middleNode(self, head: ListNode) -> ListNode:
        count = 0
        res = head
        while head:
            count += 1
            head = head.next
        mid = count // 2
        while mid:
            res = res.next
            mid -= 1
            
        return res